### DISCUSSION TITLE
mock dependency

### DISCUSSION THREAD

**Discussion by jblanckelkoo** [behave/behave#1026] [General]
Hello;

Have you planned to move to unittest.mock instead of mock? (in fedora it is deprecated)
It seems easy to migrate, ie https://fedoraproject.org/wiki/Changes/DeprecatePythonMock#How_to_migrate_to_unittest.mock

(We need to package a behave for fedora)

Thanks!

**Comment by jenisys** [comment]
Sorry, for the late reply.

While I could do that, a Fedora Python package should be a binary distribution.
The binary distribution should not contain the `tests/` subdirectory.
And the unit tests in the `tests/` subdirectory are the only ones that require the `mock` module.
Therefore, it is only listed as test dependency.

Could you explain why using the `mock` package directly is still a problem on Fedora ?

**Reply by jenisys** [reply]
Note that `mock` is only removed in `behave v1.4.0` (which is not released yet). The main reason is that `behave v1.3.x` still supports Python 2.7 and `unittest.mock` is only available in Python3. Therefore, I can not be removed in `behave v1.3.x`.
